onnxruntime/core/session/plugin_ep/ep_plugin_provider_interfaces.cc

```diff
@@ -207,27 +248,29 @@ PluginExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_vie
           this->Type(), this->Type(), /*node_unit_map*/ nullptr,
           node_grouping.fusion_options.drop_constant_initializers);
 
-      if (capabilities.size() > 1) {
-        LOGS_DEFAULT(ERROR) << "OrtEp::GetCapability() set nodes that cannot be fused together. "
-                            << "Please ensure that the nodes provided to EpGraphSupportInfo_AddFusedNodes() do not "
+      if (capabilities.size() != 1) {
+        LOGS(logger, ERROR) << "OrtEp::GetCapability() for " << Type() << " set nodes that cannot be fused together. "
+                            << "Please ensure that the nodes provided to EpGraphSupportInfo_AddNodesToFuse() do not "
                             << "have an unsupported node in any path between two of the supported nodes.";
         return {};
       }
 
-      // Enforce that the nodes in node_set match the nodes in capabilities[0]
+      // Log an error if the nodes in node_set do not match the nodes in capabilities[0]. We expect this to always
+      // be true because we've already checked that the EP did not try to claim nodes already assigned to another EP.
```

Comment:
hm, I wonder if the original `ORT_ENFORCE` checks are ok to keep. if those conditions aren't met something is not as expected, right? I suppose logging an error is ok too, but ORT_ENFORCE is louder.

Reply:
It does cost binary size though and causes an abort in a minimal build.

I'm not sure I'd pay for an abort if CreateSupportedPartitions split things up (EP might make a mistake but we can gracefully continue even if it did), or for CreateSupportedPartitions randomly adding a new node in the returned values (if we want to check that an `assert` may be a better balance).

Reply:
sure, `assert` would be fine too. the EP making that mistake should be handled with the `capabilities.size() != 1` check above. but having mismatched node sets at this point is just something that, afaik, we really don't expect to happen.